src/problem/MFEMProblem.C

```diff
@@ -172,6 +142,21 @@ void
 MFEMProblem::addVariable(const std::string & var_type,
                          const std::string & var_name,
                          InputParameters & parameters)
+{
+  addGridFunction(var_type, var_name, parameters);
+  // MOOSE variables store DoFs for the trial variable and its time derivatives up to second order;
+  // MFEM GridFunctions store data for only one set of DoFs each, so we must add additional
+  // GridFunctions for time derivatives.
+  if (isTransient())
+  {
+    addGridFunction(var_type, platypus::GetTimeDerivativeName(var_name), parameters);
+  }
+}
+
+void
+MFEMProblem::addGridFunction(const std::string & var_type,
+                             const std::string & var_name,
+                             InputParameters & parameters)
```

Comment:
It's a workaround to avoid Outputs attempting to write time derivative `GridFunctions` before they have been instantiated. 

When we were using `platypus::Outputs`, `MFEMDataCollections` would be constructed, then the `MFEMExecutioner` would be initialised (adding the time derivative `GridFunctions` for transient problems), and finally the `mfem::DataCollection` would be created, set up using all available `GridFunctions`, and registered in `platypus::Outputs`. File output was written by iterating through the registered members of `platypus::Outputs`.

In this PR, `MFEMDataCollections` are constructed ready-to-use, with file output managed via the overridden `output()` method like other MOOSE output classes. But as the construction of the `MFEMDataCollection` happens before `MFEMExecutioner::Init` call, the time derivatives gridfunctions are not available at the `MFEMDataCollection`'s construction time. 

So this change is to bring the construction of time derivative gridfunctions to be consistent with MOOSE (which creates time derivatives in the MOOSEVariable construction), avoiding this issue altogether.

I agree this would be cleaner as two separate PRs (one for the change to when time derivative `GridFunctions` are initialised, and a second dependent one on the removal of `platypus::Outputs`) - this was just exposed during the removal of `platypus::Outputs`. Happy to split them if you'd prefer?

Reply:
No, don't bother splitting them. I was just worried this was a case of you including some unrelated fix you were meaning to get to, but clearly it is not.

Reply:
OK - thanks!